api/checks/update.go

```diff
@@ -118,6 +95,41 @@ func updateCheckHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func loadRunsToCompare(ctx context.Context, filter shared.TestRunFilter) (prRun, masterRun *shared.TestRun, err error) {
+	log := shared.GetLogger(ctx)
+	one := 1
+	runs, err := shared.LoadTestRuns(ctx, filter.Products, filter.Labels, filter.SHA, filter.From, filter.To, &one, nil)
+	prRun = runs.First()
+	if err != nil {
+		log.Errorf("Failed to load test runs: %s", err.Error())
+		return nil, nil, err
+	} else if prRun == nil {
+		log.Debugf("No runs found for %s @ %s", filter.Products[0].String(), filter.SHA[:7])
+		return nil, nil, err
+	} else if len(runs.AllRuns()) > 1 {
+		log.Errorf("Found more that one test run")
+		return nil, nil, fmt.Errorf("Expected exactly 1 run, but found %v", len(runs))
+	}
+
+	// Get the most recent, but still earlier, master run to compare.
+	labels := filter.Labels
+	if labels == nil {
+		labels = mapset.NewSet()
+	}
+	labels.Add("master")
+	to := prRun.TimeStart.Add(-time.Millisecond)
```

Comment:
While not obvious in the diff, this is the actual change. The rest is a refactor so that I could add a test for it.

Reply:
Thanks for pointing that out!